mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -1047,9 +1058,14 @@ pub async fn nexus_create(
     // global variable defined in the nexus module
     let nexus_list = instances();
 
-    if nexus_list.iter().any(|n| n.name == name) {
+    if let Some(nexus) = nexus_list.iter().find(|n| n.name == name) {
         // FIXME: Instead of error, we return Ok without checking
         // that the children match, which seems wrong.
+        if *nexus.state.lock() == NexusState::Init {
+            return Err(Error::NexusNotFound {
```

Comment:
I think NexusNotFound is the incorrect error here as we have found an existing one.

Reply:
That's my fault, IIRC I changed this to NexusImcomplete later on my branch, though maybe that's still not quite right.

Reply:
@paulyoong Ok its one of the patches I applied ontop of this branch that came from you and @tiagolobocastro -- shall I fix it here? or..?